### DISCUSSION TITLE
Who is version 2 for?

### DISCUSSION THREAD

**Discussion by AndrewGretton** [mckaywrigley/chatbot-ui#1329] [Help]
Version 1 had a simple docker image that I could take, and deploy for people internally to use. We could point it at an Azure OpenAI connection and everything just worked.

Looking at version 2, there's no docker image - just a series of complex(ish) steps to set up. And it's not even clear if it's safe for multiple people to use any more - if I go through all the steps to install all the things, will users have different chat sessions or will they all share one?

Is ChatBotUI mostly just a tool for people to run themselves now, rather than to install internally for multiple people to share?

**Comment by duckworth** [comment]
I just wanted to second this sentiment. I love and appreciate all the work that has been done, but getting version 2 running in any non development capacity in my own environment has been frustrating. Supabase is an absolute nightmare to deal with and adds too much complexity to self host. Really wish there was some easier deployment options to run this thing. More than willing to help out.

**Reply by ernesttam** [reply]
I agreed. I just want a good chatbot ui and I think this one is good. However, I bumped into issues on installing supabase. I even thought about using host instance but don't know how to configure it. Sigh!

**Comment by Bortus-AI** [comment]
> Version 1 had a simple docker image that I could take, and deploy for people internally to use. We could point it at an Azure OpenAI connection and everything just worked.
> 
> Looking at version 2, there's no docker image - just a series of complex(ish) steps to set up. And it's not even clear if it's safe for multiple people to use any more - if I go through all the steps to install all the things, will users have different chat sessions or will they all share one?
> 
> Is ChatBotUI mostly just a tool for people to run themselves now, rather than to install internally for multiple people to share?

Each users has their own account and own chat sessions. I run it on a server for multiple users and no issues

**Reply by AlmostRemember** [reply]
I really want to use this tool and would like a cloud-hosted instance. I'm a professional software developer and I'm sure I could get it working, but the setup is too complex for me to bother. I don't have time to learn several new tools that I don't normally work with. Why should I have to get supabase running locally when I'm going to use a hosted database?

I'm fine with some manual steps and I don't need a one-click process, but the deployment needs to be simpler than it is.

**Reply by Bortus-AI** [reply]
@AlmostRemember you can use the free hosted supabase. You don't have to use a local supabase

**Reply by AlmostRemember** [reply]
I get that, but the hosted quickstart says you need to do the full local setup first. Am I misunderstanding something?

**Reply by gijigae** [reply]
@AlmostRemember Having Supabse CLI will help you push DB scripts from your local machine. Alternatively, you could run DB scripts from Supabase's built-in SQL Editor (see the image below). Hope it helps.
![image](https://github.com/mckaywrigley/chatbot-ui/assets/81459/6701eed3-098b-4eed-b432-76a35269ec83)

**Reply by AlmostRemember** [reply]
Thanks @gijigae, I will try this. If I get it to work maybe I'll write up my own guide.

**Reply by gijigae** [reply]
Honestly though, it will be a lot easier if you do it from Supabase CLI. Considering the current development pace, there would be frequent DB changes. With supabase CLI, you could run 'npm run db-push` to have those updates reflected in your hosted DB. Otherwise, you need to manually run updated/added scripts.

**Comment by erahhal** [comment]
Supabase defenitely makes life difficult for those of us with existing managed Postgres installations. The documentation for setting up Supabase with an existing Postgres instance is lacking to say the least, and is the biggest blocker to adoption in my case. Using the standard Supabase postgres image is not feasible as introducing new data stores with associated maintenance, uptime, and backup requirements is a non-starter.

I'm wondering how much effort it would be to add an adapter layer that allows one to swap out different data stores.

**Comment by C-Loftus** [comment]
Also having issues with the main branch. I opened an issue related to setup problems and it was closed very quickly without discussion [https://github.com/mckaywrigley/chatbot-ui/issues/1399](https://github.com/mckaywrigley/chatbot-ui/issues/1399) I know the policy is that issues unrelated to the direct repo would be closed (i.e. getting it set up on certain cloud providers) which is totally reasonable. However, this seemed to be related.

I am super thankful for all the work here, I just am a bit confused by the state of the project. It is ok to still use the legacy branch?

**Comment by AlmostRemember** [comment]
Who is version 2 for? The readme says _"The open-source AI chat app for everyone."_

It might be more accurate to say it's for a rather small group of software developers who happen to have experience with a very specific tech stack, which is perhaps not nearly as ubiquitous as the young creator assumes. 

This is the open source world so I can't really complain, but this seems closer to a hobby project than practically usable software. The fact that I'm here shows that I see the potential, but the barriers to deployment seem way too high. 

I do appreciate the effort and I'll keep checking back to see how things develop.

**Reply by Jonneal3** [reply]
@AlmostRemember Are you talking about deployment for a techie person? I had no problem deploying to vercel. Let me know if you have issues. As far as. I know, the tech stack is "the" tech stack for front-ends moving forward, speaking of React and next.js. The only other one I know making waves is svelte. If you can clarify more as to what you mean by "non-usable" or whatnot makes everyone better imo!

**Reply by AlmostRemember** [reply]
Sorry if my comment sounded harsh. I was feeling a bit frustrated. Here's what I've done so far:
1. Created a windows 11 VM sandbox to play with this
2. Installed VSCode, Git, Node.js
3. Signed in to github and cloned the code
4. Run npm install. All packages restored except for regenerator-runtime, which failed with a timeout
5. I stopped when I realized I needed to install the scoop package manager just to install the Supabase CLI (this was in my spare time)

I haven't gotten back to it yet. And I still need to get docker installed! You have to admin that's a lot of technical steps. I was hoping for something more like this for a hosted deployment:
1. Setup a hosted Supabase account and project
2. Upload the code to a hosting provider
3. Edit a config file to point to the Supabase account
4. Run a process to automatically initialize (or update) the database
5. Ready to go with nothing happening locally

**Reply by AlmostRemember** [reply]
React makes sense for an app like this where virtually everything happens on the client, but it's not the only flavor of web development going on.

I'm currently working on an enterprise healthcare app using asp.net core with a simple bootstrap UI. In that case there is back-end server processing and the client is just presentation. So no need for any javascript frameworks. 

And for client-heavy apps I've seen Blazor get a bit more adoption lately. I think running C# or other robust languages in the browser with webassembly is really interesting. It certainly has some advantages over javascript/typescript.

**Comment by erahhal** [comment]
@Jonneal3 I believe they are referring to Supabase, not the front-end stack.

**Comment by spammenotinoz** [comment]
I'm a bit of a novice, but seemed to deploy okay.
Using the hosted instance of superbase, but a small Ubuntu VPS instance to host the front-end, seems to work fine, but yes it's not point and click.

**Reply by erahhal** [reply]
Sure, I got it running with nearly zero effort on my dev machine. The question is for those who are at orgs that manage their own hosting and/or have standardized persistence layers, and supabase is a complicated multipart gearbox that adds a ton of complexity to deployments and ops in general, e.g. backups, updates, scaling, maintenance, SLAs etc. and it's not clear that supabase really adds much value on top of postgres and S3 considering the significant additional complexity.

Once again though it's an open source project so I can't complain and appreciate that it was released, and I also notice that the README was updated with a note about adding the ability to use other persistence layers.

**Comment by AndrewGretton** [comment]
Just for anyone arriving at this thread recently... I wanted to give a shout out to @mckaywrigley who's updated the README.md. It says he's looking into simpler deployment - thank you!

**Comment by redthing1** [comment]
I agree with all the above: I love chatbot-ui, I hate Supabase (with a passion). I wish we could have a simple, easily deployable docker build. I'm willing to help make it happen, even if it means a fork.

**Reply by ernesttam** [reply]
By the way, I moved on to using a few other open source Java script based UI. Surprisingly, there are quite a few available on github. Then, I used ChatGPT to help me writing simple react.js UI for myself. Now, I can have my own customized server based react.js UI.

**Reply by Bortus-AI** [reply]
> By the way, I moved on to using a few other open source Java script based UI. Surprisingly, there are quite a few available on github. Then, I used ChatGPT to help me writing simple react.js UI for myself. Now, I can have my own customized server based react.js UI.

Are you willing to share what you're using?

**Comment by spammenotinoz** [comment]
Hi curious what’s wrong with Supabase?
I now used chatgpt but when i did use this code base always deployed the front and backend in Docker containers, but still used Supabase hosted as the database.   Basically just created a Dockerfile and small optional tweaks to next.config.js
Given Supabase can also be deployed on docker, is your question how to use another database?  If so surely wouldn’t be too difficult, but honestly find hosted Supabase good.

**Reply by erahhal** [reply]
Read my comments above.
